internal/exec/util/file.go

```diff
@@ -155,20 +155,20 @@ func (u Util) WriteLink(s types.Link) error {
 	return nil
 }
 
-func (u Util) SetPermissions(f types.File) error {
-	if f.Mode != nil {
-		mode := os.FileMode(*f.Mode)
-		if err := os.Chmod(f.Path, mode); err != nil {
+func (u Util) SetPermissions(mode *int, node types.Node) error {
+	if mode != nil {
+		mode := os.FileMode(*mode)
+		if err := os.Chmod(node.Path, mode); err != nil {
 			return err
 		}
 	}
 
-	defaultUid, defaultGid, _ := getFileOwnerAndMode(f.Path)
-	uid, gid, err := u.ResolveNodeUidAndGid(f.Node, defaultUid, defaultGid)
+	defaultUid, defaultGid, _ := getFileOwnerAndMode(node.Path)
+	uid, gid, err := u.ResolveNodeUidAndGid(node, defaultUid, defaultGid)
 	if err != nil {
 		return err
 	}
-	return os.Chown(f.Path, uid, gid)
+	return os.Chown(node.Path, uid, gid)
```

Comment:
Minor: how about just catching and prefixing the error with the path here? Right now we're doing it in the `dirEntry` path on the calling side, but not in the `fileEntry` path.

Reply:
:100: This is something that's rampant in the Ignition codebase.